spark-connector/spark-common/src/main/java/com/datastrato/gravitino/spark/connector/iceberg/extensions/IcebergExtendedDataSourceV2Strategy.java

```diff
@@ -0,0 +1,246 @@
+      DropPartitionField dropPartitionField = (DropPartitionField) plan;
+      return IcebergCatalogAndIdentifier.buildCatalogAndIdentifier(
+              spark, dropPartitionField.table())
+          .map(
+              catalogAndIdentifier -> {
+                DropPartitionFieldExec dropPartitionFieldExec =
+                    new DropPartitionFieldExec(
+                        catalogAndIdentifier.catalog,
+                        catalogAndIdentifier.identifier,
+                        dropPartitionField.transform());
+                return toSeq(dropPartitionFieldExec);
+              })
+          .get();
+    } else if (plan instanceof ReplacePartitionField) {
+      ReplacePartitionField replacePartitionField = (ReplacePartitionField) plan;
+      return IcebergCatalogAndIdentifier.buildCatalogAndIdentifier(
+              spark, replacePartitionField.table())
+          .map(
+              catalogAndIdentifier -> {
+                ReplacePartitionFieldExec replacePartitionFieldExec =
+                    new ReplacePartitionFieldExec(
+                        catalogAndIdentifier.catalog,
+                        catalogAndIdentifier.identifier,
+                        replacePartitionField.transformFrom(),
+                        replacePartitionField.transformTo(),
+                        replacePartitionField.name());
+                return toSeq(replacePartitionFieldExec);
+              })
+          .get();
+    } else if (plan instanceof SetIdentifierFields) {
+      SetIdentifierFields setIdentifierFields = (SetIdentifierFields) plan;
+      return IcebergCatalogAndIdentifier.buildCatalogAndIdentifier(
+              spark, setIdentifierFields.table())
+          .map(
+              catalogAndIdentifier -> {
+                SetIdentifierFieldsExec setIdentifierFieldsExec =
+                    new SetIdentifierFieldsExec(
+                        catalogAndIdentifier.catalog,
+                        catalogAndIdentifier.identifier,
+                        setIdentifierFields.fields());
+                return toSeq(setIdentifierFieldsExec);
+              })
+          .get();
+    } else if (plan instanceof DropIdentifierFields) {
+      DropIdentifierFields dropIdentifierFields = (DropIdentifierFields) plan;
+      return IcebergCatalogAndIdentifier.buildCatalogAndIdentifier(
+              spark, dropIdentifierFields.table())
+          .map(
+              catalogAndIdentifier -> {
+                DropIdentifierFieldsExec dropIdentifierFieldsExec =
+                    new DropIdentifierFieldsExec(
+                        catalogAndIdentifier.catalog,
+                        catalogAndIdentifier.identifier,
+                        dropIdentifierFields.fields());
+                return toSeq(dropIdentifierFieldsExec);
+              })
+          .get();
+    } else if (plan instanceof SetWriteDistributionAndOrdering) {
+      SetWriteDistributionAndOrdering setWriteDistributionAndOrdering =
```

Comment:
is it possible set distribution and order though Gravitino?

Reply:
Now  have not any other way to set distribution and ordering of iceberg table based on sql.

![image](https://github.com/datastrato/gravitino/assets/94670132/5e1e161e-eaa0-4cca-be54-86b54025304c)

Reply:
got it